xdsl/dialects/builtin.py

```diff
@@ -1927,6 +1927,10 @@ def get_shape(self) -> tuple[int, ...]:
     def get_element_type(self) -> IntegerType | IndexType | AnyFloat:
         return self.type.get_element_type()
 
+    @property
+    def nb_elements(self) -> int:
+        return prod(self.get_shape())
+
```

Comment:
The code changes are now looking great. Just to make sure that I didn't miss anything, could you please split this PR into one that introduces this helper and a verification to check that the number of elements contained matches the size of the shape? The goal here is for this PR to change only the underlying implementation and none of the API.

Also, I think this could just be `__len__`

Reply:
okay, I will do that, thank you for the review!